sos/plugins/cnv.py

```diff
@@ -0,0 +1,234 @@
+            if line.startswith("NAMESPACE") \
+                    or line.startswith("No resources found") \
+                    or line.startswith("Error") \
+                    or line.startswith("error"):
+                pass
+            else:
+                if line.split()[1] == option:
+                    out.append(line.split()[1] + " -n " + line.split()[0])
+        return out
+
+    def get_node_info(self, pods):
+        out = ""
+        try:
+            node_out = open(pods).read()
+        except IOError:
+            return out
+        for line in node_out.splitlines():
+            if line.startswith("NAMESPACE") \
+                    or line.startswith("No resources found") \
+                    or line.startswith("Error") \
+                    or line.startswith("error"):
+                pass
+            else:
+                out = line.split()[7]
+        return out
+
+    def setup(self):
+        oc_admin_cfg = os.path.join(self.master_base_dir, "admin.kubeconfig")
+        oc_cmd = "%s --config=%s" % ("oc", oc_admin_cfg)
+
+        all_dep = [
+            "%s get deployment -l kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_svc = [
+            "%s get service -l kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get service -l cdi.kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_pods = [
+            "%s get pods -l kubevirt.io=virt-controller "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l kubevirt.io=virt-api "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l cdi.kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l app=containerized-data-importer "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_stor = [
+            "%s get pvc --all-namespaces -o wide" % oc_cmd,
+        ]
+
+        subcmds = [
+            "nodes",
+            "clusterroles",
+            "clusterrolebinding",
+            "serviceaccount --all-namespaces",
+            "storageclass --all-namespaces",
+            "pv --all-namespaces",
+            "dv --all-namespaces",
+        ]
+        subcmds_dep = [
+            "describe deployment",
+        ]
+        subcmds_svc = [
+            "describe svc",
+        ]
+        subcmds_pods = [
+            "describe pod",
+            "logs",
+        ]
+
+        self.add_cmd_output(["%s version" % oc_cmd])
+        self.add_cmd_output(["virtctl --kubeconfig %s version" % oc_admin_cfg])
+        for s in subcmds:
+            self.add_cmd_output(["%s " % oc_cmd + "get " + s])
+        for dep in all_dep:
+            for d in self.get_resc_ns(self.get_cmd_output_now(dep)):
+                for s in subcmds_dep:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + d])
+        for svc in all_svc:
```

Comment:
Same here - they're generic kubernetes resources, and in my opinion should be folded in therre.